app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -132,34 +132,43 @@ class ConversationViewModel @Inject constructor(
     }
 
     private fun listenConversationDetails() = viewModelScope.launch {
-        observeConversationDetails(conversationId)
-            .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>() // TODO handle StorageFailure
-            .map { it.conversationDetails }
-            .collect { conversationDetails ->
-                val conversationName = when (conversationDetails) {
-                    is ConversationDetails.OneOne -> conversationDetails.otherUser.name.orEmpty()
-                    else -> conversationDetails.conversation.name.orEmpty()
-                }
-                val conversationAvatar = when (conversationDetails) {
-                    is ConversationDetails.OneOne ->
-                        ConversationAvatar.OneOne(conversationDetails.otherUser.previewPicture?.let {
-                            UserAvatarAsset(wireSessionImageLoader, it)
-                        })
-                    is ConversationDetails.Group ->
-                        ConversationAvatar.Group(conversationDetails.conversation.id)
-                    else -> ConversationAvatar.None
-                }
-                val conversationDetailsData = when (conversationDetails) {
-                    is ConversationDetails.Group -> ConversationDetailsData.Group(conversationDetails.conversation.id)
-                    is ConversationDetails.OneOne -> ConversationDetailsData.OneOne(conversationDetails.otherUser.id)
-                    else -> ConversationDetailsData.None
+        withContext(dispatchers.io()) {
+            observeConversationDetails(conversationId)
+                .collect { result ->
+                    when (result) {
+                        is Failure -> navigateToHome()
```

Comment:
Niceee. I didn't know the moment a group is deleted, this use case returns a `Failure` result. That definitely makes this look way cleaner 🧹 👏

Reply:
Small question, did you try deleting the group while the app is in airplane mode or w/o connection? would this navigate also back to home?

Reply:
Yup I tried. We have to consider that this only takes care for the receiving end of a deleted convo event. 

So if we are on airplane/no connection mode, the app will do nothing, but when gaining connection, we will start processing pending events again, having this flow correctly navigated back to home.

Reply:
I have the same concern - we do not recognize the type of error here so will it also navigate to "home" when there is for example `NoNetworkConnection` failure? 
What about `StorageFailure`? There was a TODO about handling it and probably should be handled differently than navigating to "home" 😄